javalib/src/main/scala/java/util/LinkedList.scala

```diff
@@ -0,0 +1,260 @@
+    with List[E] with Deque[E] with Cloneable with Serializable { self =>
+
+  def this(c: Collection[_ <: E]) = {
+    this()
+    addAll(c)
+  }
+
+  val inner: mutable.ListBuffer[Box[E]] = new mutable.ListBuffer()
+
+  def getFirst(): E =
+    if (inner.isEmpty) throw new NoSuchElementException()
+    else inner.head.inner
+
+  def getLast(): E =
+    if (inner.isEmpty) throw new NoSuchElementException()
+    else inner.last.inner
+
+  def removeFirst(): E =
+    if (inner.isEmpty) throw new NoSuchElementException()
+    else inner.remove(0).inner
+
+  def removeLast(): E =
+    if (inner.isEmpty) throw new NoSuchElementException()
+    else inner.remove(inner.size - 1).inner
+
+  def addFirst(e: E): Unit =
+    inner.+=:(Box(e))
+
+  def addLast(e: E): Unit =
+    inner += Box(e)
+
+  override def contains(o: Any): Boolean = {
+    val boxed = Box(o.asInstanceOf[E])
+    inner.contains(boxed)
+  }
+
+  override def size(): Int =
+    inner.size
+
+  override def add(e: E): Boolean = {
+    addLast(e)
+    true
+  }
+
+  override def remove(o: Any): Boolean = {
+    val sizeBefore = inner.size
+    inner -= Box(o.asInstanceOf[E])
+    sizeBefore!= inner.size
+  }
+
+  override def addAll(c: Collection[_ <: E]): Boolean = {
+    val iter = c.iterator
+    var changed = false
+    while (iter.hasNext()) {
+      addLast(iter.next())
+      changed = true
+    }
+    changed
+  }
+
+  override def clear(): Unit =
+    inner.clear()
+
+  override def get(index: Int): E =
+    if (index < 0 || index >= size()) throw new IndexOutOfBoundsException(index.toString)
+    else inner(index).inner
+
+  override def set(index: Int, element: E): E = {
+    add(index, element)
+    remove(index+1)
+  }
+
+  override def add(index: Int, element: E): Unit =
+    if (index < 0 || index > size()) throw new IndexOutOfBoundsException(index.toString)
+    else inner.insert(index, Box(element))
+
+  override def remove(index: Int): E =
+    if (index < 0 || index >= size()) throw new IndexOutOfBoundsException(index.toString)
+    else inner.remove(index).inner
+
+  override def indexOf(o: Any): Int =
+    inner.indexOf(Box(o.asInstanceOf[E]))
+
+  override def lastIndexOf(o: Any): Int =
+    inner.lastIndexOf(Box(o.asInstanceOf[E]))
+
+  def peek(): E =
+    peekFirst()
+
+  def element(): E =
+    getFirst()
+
+  def poll(): E =
+    pollFirst()
+
+  def remove(): E =
+    remove(0)
+
+  def offer(e: E): Boolean =
+    offerLast(e)
+
+  def offerFirst(e: E): Boolean = {
+    val sizeBefore = size()
+    addFirst(e)
+    size() > sizeBefore
+  }
+
+  def offerLast(e: E): Boolean = {
+    val sizeBefore = size()
+    addLast(e)
+    size() > sizeBefore
+  }
+
+  def peekFirst(): E =
+    if (inner.isEmpty) null.asInstanceOf[E]
+    else inner.head.inner
+
+  def peekLast(): E =
+    if (inner.isEmpty) null.asInstanceOf[E]
+    else inner.last.inner
+
+  def pollFirst(): E = {
+    val ret = peekFirst()
+    if (ret == null) null.asInstanceOf[E] else {
```

Comment:
`LinkedList` permits `null` elements. In the case where you `peekFirst` on a `null` element, `inner` will not be trimmed. You can use `inner.headOption` to avoid this ambiguity on `null` values.